### ISSUE TITLE
Empty output directory and .json file

### ISSUE THREAD

**Issue by jmtyszka** [jmtyszka/bidskit#3]
For our project we would like to share our MRI data and therefore put the data into BIDS-format. 
We tried to use the command dcm2bids.py in the terminal on our computer (ubuntu with python 3). 
However the output is not the output that we were expecting.
The command makes an output folder and a .json file but both are empty. 
Could you help us to find the reason why both folder and file are empty? 

Kelly Berckmans - WP
Dept of Psychology and data analysis
Henri Dunantlaan 1
BE-9000 Ghent Belgium​
kelly.berckmans@ugent.be

**Comment by haerts** [comment]
Dear Mike,

Thank you for your quick reply. Kelly and I have tried the things you sent us, however with little success. 

1. Make sure you have the latest versions of both bidskit and dcm2niix. Use "git clone https://github.com/jmtyszka/bidskit.git” for bidskit and re-download and install MRIcroGL (https://www.nitrc.org/projects/mricrogl/)
--> Okay: tried git pull in bidskit folder and was already up-to-date, re-installed MRIcroGL. 

2. Make sure you’re not using compressed DICOM files, which aren’t handled well by dcm2niix.
--> If I run dcm2niix directly on the data it works fine, so this shouldn't be the problem.

3. Make sure that your source DICOM folder is organized in the following hierarchy: subject, session, DICOM files (either with or without further hierarchy, such as series). 
--> Our data is organized as follows:

CON_002
    └── POST_OP
        ├── subjectname_filenumber.IMA
        ├── subjectname_filenumber.IMA
        ...
    └── PRE_OP
        ...
For now, we're just trying to get it running on 1 subject. We're working with Siemens DICOM files (.IMA), could this have anything to do with the problem?

When we run "dcm2bids.py -i /path/CON_002/ -o /path/NEW_BIDS_CON_002/", we still have the same error. In particular, we get an output folder and a .json file, but both are empty.

Do you have some more suggestions we could try to get this working?

Thank you!

Best regards,
Hannelore & Kelly

**Comment by jmtyszka** [comment]
Thanks for running through these steps. I wanted to rule out simple issues before diving deeper.
I haven't tested dcm2bids with .IMA exports yet, so I'll try this today. What software version is your scanner running (we have VB17A Tim Trios here)?

**Comment by jmtyszka** [comment]
I just tested dcm2bids locally with a QA phantom run exported in Siemens .IMA format. The IMA files were organized as follows:

> mydicom
> └── QA_CBIC_QA_20170613_090054_437000
>     └── FBIRN_STABILITY_0001
>         ├── QA_20170613_32.MR.QA_CBIC_QA.0001.0001.2017.06.14.09.32.15.0.98718896.IMA
>         ├── QA_20170613_32.MR.QA_CBIC_QA.0001.0002.2017.06.14.09.32.15.0.98718911.IMA
>         ├── QA_20170613_32.MR.QA_CBIC_QA.0001.0003.2017.06.14.09.32.15.0.98718926.IMA
>         ├── QA_20170613_32.MR.QA_CBIC_QA.0001.0004.2017.06.14.09.32.15.0.98718941.IMA
>         ├── QA_20170613_32.MR.QA_CBIC_QA.0001.0005.2017.06.14.09.32.15.0.98718956.IMA
> ...
> 

And the initial output from dcm2bids.py looked like this:

> evendim:IMA_Test jmt$ dcm2bids.py -i mydicom -o mybids
> 
> ------------------------------------------------------------
> Pass 1 : DICOM to Nifti conversion and dictionary creation
> ------------------------------------------------------------
> 
> Processing subject QA_CBIC_QA_20170613_090054_437000
>   Processing session FBIRN_STABILITY_0001
>   Converting all DICOM images within directory mydicom/QA_CBIC_QA_20170613_090054_437000/FBIRN_STABILITY_0001/
>   Adding protocol fBIRN_Stability to dictionary template
>   Adding protocol fBIRN_Stability to dictionary template
> 
> ---
> New protocol dictionary created : mydicom/Protocol_Translator.json
> Remember to replace "EXCLUDE" values in dictionary with an appropriate image description
> For example "MP-RAGE T1w 3D structural" or "MB-EPI BOLD resting-state
> ---

I've added a small zip archive to the repo (mydicom.zip) containing this phantom IMA-format test data with the above directory structure. See if running "dcm2bids.py -i mydicom -o mybids" on this data works at your end. If it does, then all you probably need to do is put your existing "CON_002" directory within an enclosing folder (eg "mydicom" or an appropriate dataset name) and run dcm2bids on this folder.

Let me know if this works.

**Comment by haerts** [comment]
Dear Mike,

I created one extra level up the hierarchy "mydicom" as you suggested, including the one subject CON02, and now the first step works! 

hannelore@Hannelore:~/Kelly_BIDS$ dcm2bids.py -i mydicom/ -o mybids/
------------------------------------------------------------
Pass 1 : DICOM to Nifti conversion and dictionary creation
------------------------------------------------------------
Processing subject CON02
  Processing session PRE_OP
  Converting all DICOM images within directory mydicom/CON02/PRE_OP/
  Adding protocol RESTING_STATE_6min to dictionary template
  Adding protocol T1_mprage to dictionary template
  Adding protocol DTI_2.5_b2800_101dir to dictionary template
  Adding protocol PA_2averages_MDDW_6dir to dictionary template
  Processing session POST_OP
  Converting all DICOM images within directory mydicom/CON02/POST_OP/
  Adding protocol T1_mprage to dictionary template
  Adding protocol DTI_2.5_b2800_101dir to dictionary template
  Adding protocol fMRI_Resting_State to dictionary template
  Adding protocol PA_2averages_MDDW_6dir to dictionary template
---
New protocol dictionary created : mydicom/Protocol_Translator.json
Remember to replace "EXCLUDE" values in dictionary with an appropriate image description
For example "MP-RAGE T1w 3D structural" or "MB-EPI BOLD resting-state
---

I adapted the Protocol_Translator.json file and reran dcm2bids.py. However, now I get this error: 

hannelore@Hannelore:~/Kelly_BIDS$ dcm2bids.py -i mydicom -o mybids
------------------------------------------------------------
Pass 2 : Organizing Nifti data into BIDS directories
------------------------------------------------------------
Processing subject CON02
  Processing session PRE_OP
* No DICOM header information found in mydicom/CON02/PRE_OP/
* Confirm that DICOM images in this folder are uncompressed
* Exiting

The only thing that happened is that a "dataset_description.json" and a "participants.tsv" file were created. Nevertheless, dicom information is present in this folder:

hannelore@Hannelore:~/Kelly_BIDS/mydicom/CON03/PRE_OP$ **dcminfo** DWI/GIFMI_STUCONN_SUB007P_.MR.KLINIEK_3T_HOOFD_HERSENEN_32KANAALS_HC.0005.0102.2015.11.09.14.59.56.617453.7301400.IMA 
file: "DWI/GIFMI_STUCONN_SUB007P_.MR.KLINIEK_3T_HOOFD_HERSENEN_32KANAALS_HC.0005.0102.2015.11.09.14.59.56.617453.7301400.IMA" [MR]:
    patient: GIFMI STUCONN SUB007p   (QP-1388010) - 09/11/1962
    study: Kliniek 3T Hoofd Hersenen 32kanaals HC (3096746) - 09/11/2015 14:23:13
    series: [5] DTI_2.5_b2800_101dir - 09/11/2015 14:46:43
    sequence: *ep_b0

Do you have an idea what might have gone wrong?

Thanks!

**Comment by jmtyszka** [comment]
Not sure what's going on here, but I've updated dcm2bids.py to use the latest master version of dcm2niix from Chris Rorden's repo (see README for this repo). I'd suggest switching to the rordenlab version of dcm2niix, pulling bidskit again and rerunning the second pass dcm2bids.py (ie don't delete your output BIDS directory or Protocol_Translator.json file). Let me know if this fixes anything.

**Comment by haerts** [comment]
Hi Mike,

I did as you suggested (downloaded dcm2niix from rordenlab and again cloned the bidskit repo) and in addition used Python 2.7 (instead of 3.4) and now it works just fine! Which is a bit weird, since you explicitly say it's a python3 function? I saw the explanation (describing it as a python3 function) only after giving it a try with python2. The output seems fine though... Can I then just use it with python2?

Thanks!
